### PR TITLE
Support wazuh-logcollector tests

### PR DESCRIPTION
**Pull request by Nicogp**
|Related Issue |
|-----------------------|
|  wazuh/wazuh#19085  |

## Description

This PR includes all the changes necessary to run `wazuh-logcollector` tests:

- Module `wazuh-logcollector` with all the necessary constants.
- Util `validate_test_config_with_module_config` to validate the configuration applied in the test with the configuration returned by the getconfig request (via internal socket).

### CODE REVIEW

File: src/wazuh_testing/modules/logcollector/utils.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2015-2024, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import json
+from wazuh_testing.constants.paths.sockets import LOGCOLLECTOR_SOCKET_PATH
+from wazuh_testing.utils import sockets
+from wazuh_testing.modules.logcollector import patterns
+
```
Comment (TomasTurina) [inline src/wazuh_testing/modules/logcollector/utils.py:9]: Missing empty space.

File: src/wazuh_testing/modules/logcollector/patterns.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ERROR_LOG_FILE_DUPLICATED = 'The expected warning log file duplicated has not been produced.'
+ERROR_ANALYZING_MACOS = 'The expected analyzing macos log has not been produced'
+ERROR_DJB_MULTILOG_NOT_PRODUCED = 'The expected multilog djb log has not been produced'
+ERROR_INVALID_MACOS_VALUE = 'The expected warning invalid macos value has not been produced'
+ERROR_MISSING_LOCATION_VALUE = 'The expected warning missing location value has not been produced'
+ERROR_MACOS_LOG_NOT_PRODUCED = 'The expected macos log monitoring has not been produced'
+ERROR_EVENTCHANNEL = 'Did not receive the expected "ERROR: Could not EvtSubscribe() for ... which returned ... event.'
+ERROR_INVALID_RECONNECTION_TIME = 'The expected invalid reconnection time error has not been produced'
+
+START_UP_TIMEOUT = 10
```
Comment (TomasTurina) [inline src/wazuh_testing/modules/logcollector/patterns.py:40]: I think these 2 constants should be in the tests, not here.

File: src/wazuh_testing/modules/logcollector/patterns.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+LOG_COLLECTOR_GLOBAL_TIMEOUT = 40
+
+MACOS_LOG_COMMAND_PATH = '/usr/bin/log'
```
Comment (TomasTurina) [inline src/wazuh_testing/modules/logcollector/patterns.py:43]: I think the right place for this is `src/wazuh_testing/constants/paths/logs.py`.

File: src/wazuh_testing/modules/logcollector/patterns.py (hunk 2 above)
Comment (Nicogp) [reply src/wazuh_testing/modules/logcollector/patterns.py]: Done

File: src/wazuh_testing/modules/logcollector/patterns.py (hunk 3 above)
Comment (Nicogp) [reply src/wazuh_testing/modules/logcollector/patterns.py]: Done

File: src/wazuh_testing/modules/logcollector/utils.py (hunk 1 above)
Comment (Nicogp) [reply src/wazuh_testing/modules/logcollector/utils.py:12]: Done

**Review by TomasTurina — DISMISSED** [review]

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**